src/Mod/PartDesign/App/FeaturePocket.cpp

```diff
@@ -141,7 +150,78 @@ App::DocumentObjectExecReturn *Pocket::execute(void)
 
         base.Move(invObjLoc);
 
-        gp_Dir dir(SketchVector.x,SketchVector.y,SketchVector.z);
+        Base::Vector3d pocketDirection;
+
+        if (!UseCustomVector.getValue()) {
+            if (ReferenceAxis.getValue() == nullptr) {

```

Comment:
Why switch syntax between these two conditionals? Core Guidelines prefer your style on 155 to the style on 156, but is there some reason to really highlight the `nullptr` comparison on 156?

Reply:
The first one is a bool, the second one a pointer therefore I checked for a nullptr to check that there is no axis. What are the guidelines in this case?
(I really miss a defined guideline that I can check myself.)

Reply:
I'm referring to: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if

Reply:
> I'm referring to: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if

Many thanks. So can we just use these guidelines for FC? Should I start a forum thread to make this decision?

I improved the code now accordingly.

Reply:
I meant to make the style guideline mandatory. I remember we had a discussion this year and this guideline was mentioned by Werner as the one he follows too. But it would be good to have this info for new developers.

So what I can do is to add it to the Wiki and writ to the forum. OK?